src/app/games/diaballik/tests/DiaballikMove.spec.ts

```diff
@@ -0,0 +1,133 @@
+import { EncoderTestUtils } from 'src/app/utils/tests/Encoder.spec';
+import { DiaballikMove, DiaballikPass, DiaballikTranslation } from '../DiaballikMove';
+import { Coord } from 'src/app/jscaip/Coord';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { MGPFallible } from 'src/app/utils/MGPFallible';
+import { RulesUtils } from 'src/app/jscaip/tests/RulesUtils.spec';
+import { DiaballikFailure } from '../DiaballikFailure';
+
+describe('DiaballikMove', () => {
+    it('should reject out of board moves', () => {
+        const invalidCoord: Coord = new Coord(-1, 0);
+        const validCoord: Coord = new Coord(0, 0);
+        RulesUtils.expectToThrowAndLog(() => DiaballikTranslation.from(invalidCoord, validCoord),
+                                       'DiaballikMove not on board');
+        RulesUtils.expectToThrowAndLog(() => DiaballikTranslation.from(validCoord, invalidCoord),
+                                       'DiaballikMove not on board');
+        RulesUtils.expectToThrowAndLog(() => DiaballikPass.from(invalidCoord, validCoord),
+                                       'DiaballikMove not on board');
+        RulesUtils.expectToThrowAndLog(() => DiaballikPass.from(validCoord, invalidCoord),
+                                       'DiaballikMove not on board');
+    });
+    it('should reject move with more than one pass', () => {
+        const pass: DiaballikPass = DiaballikPass.from(new Coord(0, 0), new Coord(1, 0)).get();
+        const translation: DiaballikTranslation = DiaballikTranslation.from(new Coord(0, 0), new Coord(1, 0)).get();
+        RulesUtils.expectToThrowAndLog(() => new DiaballikMove(pass, MGPOptional.of(pass), MGPOptional.empty()),
+                                       'DiaballikMove should have at most one pass');
+        RulesUtils.expectToThrowAndLog(() => new DiaballikMove(pass, MGPOptional.of(translation), MGPOptional.of(pass)),
+                                       'DiaballikMove should have at most one pass');
+    });
+    it('should reject move with three translations', () => {
+        const first: DiaballikTranslation = DiaballikTranslation.from(new Coord(0, 0), new Coord(0, 1)).get();
+        const second: DiaballikTranslation = DiaballikTranslation.from(new Coord(0, 1), new Coord(1, 1)).get();
+        const third: DiaballikTranslation = DiaballikTranslation.from(new Coord(1, 1), new Coord(2, 1)).get();
+        RulesUtils.expectToThrowAndLog(() => new DiaballikMove(first, MGPOptional.of(second), MGPOptional.of(third)),
+                                       'DiaballikMove should have at most two translations');
+
+
+    });
```

Comment:
\n\n\n\n

Reply:
en vrai en profiter pour taper du \n entre chaque it !

Reply:
veut-on vraiment du \n entre chaque it? Cf. discussion plus haut

Reply:
4 > 11 ?
Si c'est ça eu bah, con fer toi même :D !

Reply:
fait !